code/modules/clothing/modular_armor/attachments/badges.dm

```diff
@@ -0,0 +1,111 @@
+		"V (Right Chest)" = /datum/greyscale_config/badge/rchest/V,
+		"V (Left Arm)" = /datum/greyscale_config/badge/larm/V,
+		"V (Right Arm)" = /datum/greyscale_config/badge/rarm/V,
+		"V (Beret Forward)" = /datum/greyscale_config/badge/beretf/V,
+		"V (Beret Side)" = /datum/greyscale_config/badge/berets/V,
+		"V (Left Helm)" = /datum/greyscale_config/badge/lhelm/V,
+		"V (Right Helm)" = /datum/greyscale_config/badge/rhelm/V,
+
+		"Lambda (Left Chest)" = /datum/greyscale_config/badge/lchest/lambda,
+		"Lambda (Right Chest)" = /datum/greyscale_config/badge/rchest/lambda,
+		"Lambda (Left Arm)" = /datum/greyscale_config/badge/larm/lambda,
+		"Lambda (Right Arm)" = /datum/greyscale_config/badge/rarm/lambda,
+		"Lambda (Beret Forward)" = /datum/greyscale_config/badge/beretf/lambda,
+		"Lambda (Beret Side)" = /datum/greyscale_config/badge/berets/lambda,
+		"Lambda (Left Helm)" = /datum/greyscale_config/badge/lhelm/lambda,
+		"Lambda (Right Helm)" = /datum/greyscale_config/badge/rhelm/lambda,
+
+		"Circle Two (Left Chest)" = /datum/greyscale_config/badge/lchest/circle2,
+		"Circle Two (Right Chest)" = /datum/greyscale_config/badge/rchest/circle2,
+		"Circle Two (Left Arm)" = /datum/greyscale_config/badge/larm/circle2,
+		"Circle Two (Right Arm)" = /datum/greyscale_config/badge/rarm/circle2,
+		"Circle Two (Beret Forward)" = /datum/greyscale_config/badge/beretf/circle2,
+		"Circle Two (Beret Side)" = /datum/greyscale_config/badge/berets/circle2,
+		"Circle Two (Left Helm)" = /datum/greyscale_config/badge/lhelm/circle2,
+		"Circle Two (Right Helm)" = /datum/greyscale_config/badge/rhelm/circle2,
+	)
+
+/obj/item/armor_module/armor/badge/Initialize()
+	. = ..()
+	style_list = string_list(style_list)
+	item_state = initial(icon_state)
+	update_icon()
+
+/obj/item/armor_module/armor/badge/limit_colorable_colors(faction)
+	return
+
+/obj/item/armor_module/armor/can_attach(obj/item/attaching_to, mob/user)
+	. = ..()
+	var/allowed = TRUE
+	if((istype(attaching_to, /obj/item/clothing/under) || istype(attaching_to, /obj/item/clothing/suit)) && (!ispath(greyscale_config, /datum/greyscale_config/badge/lchest) && !ispath(greyscale_config, /datum/greyscale_config/badge/rchest) && !ispath(greyscale_config, /datum/greyscale_config/badge/larm) && !ispath(greyscale_config, /datum/greyscale_config/badge/rarm)))
+		allowed = FALSE
+	if((istype(attaching_to, /obj/item/clothing/head/beret) || istype(attaching_to, /obj/item/clothing/head/tgmcberet)) && (!ispath(greyscale_config, /datum/greyscale_config/badge/beretf) && !ispath(greyscale_config, /datum/greyscale_config/badge/berets)))
+		allowed = FALSE
+	if((istype(attaching_to, /obj/item/clothing/head) && !istype(attaching_to, /obj/item/clothing/head/beret && !istype(attaching_to, /obj/item/clothing/head/tgmcberet))) && (!ispath(greyscale_config, /datum/greyscale_config/badge/lhelm) && !ispath(greyscale_config, /datum/greyscale_config/badge/rhelm)))
+		allowed = FALSE
```

Comment:
wtf is this hell check

Reply:
Its checking the currently selected sytle (aka greyscale datum) and checking it against the type you are attaching it to. No beret style badges attached to uniforms